scaletest/workspacetraffic/run_test.go

```diff
@@ -261,9 +264,102 @@ func TestRun(t *testing.T) {
 		t.Logf("bytes written total: %.0f\n", writeMetrics.Total())
 
 		// We want to ensure the metrics are somewhat accurate.
-		// TODO: https://github.com/coder/coder/issues/11175
-		// assert.InDelta(t, bytesPerTick, writeMetrics.Total(), 0.1)
+		assert.InDelta(t, readMetrics.Total(), writeMetrics.Total(), float64(bytesPerTick))
+		// Read is highly variable, depending on how far we read before stopping.
+		// Just ensure it's not zero.
+		assert.NotZero(t, readMetrics.Total())
+		// Latency should report non-zero values.
+		assert.NotEmpty(t, readMetrics.Latencies())
+		assert.NotEmpty(t, writeMetrics.Latencies())
+		// Should not report any errors!
+		assert.Zero(t, readMetrics.Errors())
+		assert.Zero(t, writeMetrics.Errors())
+	})
+
+	t.Run("App", func(t *testing.T) {
+		t.Parallel()
+
+		// Start a test server that will echo back the request body, this skips
+		// the roundtrip to coderd/agent and simply tests the http request conn
+		// directly.
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			c, err := websocket.Accept(w, r, &websocket.AcceptOptions{})
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			nc := websocket.NetConn(context.Background(), c, websocket.MessageBinary)
+			defer nc.Close()
+
+			_, err = io.Copy(nc, nc)
+			if err != nil && !errors.Is(err, io.EOF) {
+				t.Error(err)
+				return
```

Comment:
don't need return here

Reply:
I know it's not explicitly needed, but if code is added underneath then it can become problematic. I can remove it if you prefer, though.